WIP: GeoPose encoding

anermina opened the pull request:
As requested in #1 .

anermina committed:
Encode GeoPose document

anermina committed:
Modify markup

anermina commented:
@manuelfuenmayor could you please generate a document based on the uploaded files and double-check how nested requirements are shown in the corresponding requirements class? I get this strange numbering format:
![image](https://user-images.githubusercontent.com/55683568/178121623-00e6be8e-3aae-4643-9d26-a93b1c94691b.png)

Thank you in advance!

manuelfuenmayor committed:
Chore: Link Conformance class/Conformance test to Requirement class/Requirement

manuelfuenmayor committed:
Correct label of "GP-Instant-SDU" requirement

manuelfuenmayor committed:
Correct id values

manuelfuenmayor commented:
@anermina yes, I get the same result. But I think this is expected given that the
macro uses the number of the requirement instead of the name. For example,
"Requirement 34" is rendered as:
![capture2](https://user-images.githubusercontent.com/33627611/178128622-0e6a2186-b026-4f4c-a4f4-b9ceaf179675.PNG)

Unlike original, that includes the name of the requirement instead of the 
number:
![capture3](https://user-images.githubusercontent.com/33627611/178128659-acb1fc99-c066-4809-8e09-26a1a32f8171.PNG)

In the case of "Requirement 4-2", I think there is an error from original source.
The label should be `/req/time/geopose_instant/sdu` instead of `/req/geopose_instant/sdu`.
To match with this requirement:
![capture4](https://user-images.githubusercontent.com/33627611/178128720-d9fa2d1f-7ef9-473f-a12f-1ec64b942fcf.PNG)

Maybe the field "time" was omitted by error. (I've corrected it.)

In addition, I have applied a markup update to link some conformance tests
to their respective requirements.

anermina commented:
Thanks @manuelfuenmayor ! 

This document is a draft, so there are some errors in the original source files. I've also applied some corrections:
* "Requirement" field in "Requirement S-FT-Element-SDU" and "Requirement S-Trailer-SDU" has been encoded as `description` - it seems to me there is an error in the original document, which shows description under ´Requirement´.
* I changed a typo in "Implemntation" to "Implementation" throughout the text.

After double-checking the document, do you think we're good to go with this PR?

Note: I kept the original form of the files and folders, although we usually have a separate subfolder for sections and one subfolder for images.

manuelfuenmayor committed:
Chore: Divide long paragraphs into smaller lines

manuelfuenmayor commented:
> After double-checking the document, do you think we're good to go with this PR?

@anermina there are still many "conformance test to requirement" links to make:

````
Requirements: (XML Line 003570): Conformance test rta1 has no corresponding Requirement
Requirements: (XML Line 003595): Conformance test rta2 has no corresponding Requirement
Requirements: (XML Line 003621): Conformance test rta3 has no corresponding Requirement
...
````

...but I haven't been able to find the correct markup to do so. I have opened a [ticket](https://github.com/metanorma/ogc-GeoPose/issues/3)
asking for help. Maybe we should keep this PR opened a little longer.
Thanks!

manuelfuenmayor commented:
> Note: I kept the original form of the files and folders, although we usually have a separate subfolder for sections and one subfolder for images.

@ronaldtse what do you think about this?
Should we maintain the original form of files and folders?

ronaldtse commented:
> @anermina there are still many "conformance test to requirement" links to make:
> 
> ```
> Requirements: (XML Line 003570): Conformance test rta1 has no corresponding Requirement
> Requirements: (XML Line 003595): Conformance test rta2 has no corresponding Requirement
> Requirements: (XML Line 003621): Conformance test rta3 has no corresponding Requirement
> ...
> ```
> 
> ...but I haven't been able to find the correct markup to do so. I have opened a [ticket](https://github.com/metanorma/ogc-GeoPose/issues/3) asking for help. Maybe we should keep this PR opened a little longer. Thanks!

The correct markup is shown here:
https://www.metanorma.org/author/ogc/topics/requirements/

If the syntax is correctly followed it should work.

ronaldtse commented on standard/standard/annex-bibliography.adoc:
```diff
@@ -1,38 +1,34 @@
+* [[[geometric-tools,2]]], PHILIP J. SCHNEIDER, DAVID H. EBERLY, Geometric Tools for Computer Graphics, Morgan Kaufmann, 2003.
 
-[2] PHILIP J. SCHNEIDER, DAVID H. EBERLY, Geometric Tools for Computer Graphics, Morgan Kaufmann, 2003.
+* [[[ogc-testbed14,3]]], OGC Testbed-14: CityGML and AR Engineering Report ( https://docs.ogc.org/per/18-025.html )
```
Change to use auto-fetch.

manuelfuenmayor replied:
Done.

ronaldtse commented on standard/standard/annex-bibliography.adoc:
```diff
@@ -1,38 +1,34 @@
-:appendix-caption: Annex
-[[Bibliography]]
-= Bibliography
+[[annex-bibliography]]
```
Please do not give anchors to "bibliographies", no one needs to cite them.

manuelfuenmayor replied:
Ok, fixed.

ronaldtse replied:
Thanks.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1279 @@
+[conformance_test,id="rta1"]
+======
+[%metadata]
+label:: <<conformance_global,/conf/target_dependency>>
```
A 'label' is a string, an identifier. Why is it a cross-reference?

manuelfuenmayor replied:
Because in original source the ID was marked up as a cross-reference.

From original markup:
```adoc
[[conf_global]]
[width="100%",%header,cols="2,6"]
|===
<|*Conformance Class:* <|*Global* 

<s|ID |<<conf_global,/conf/global>>
...
|===
````

ronaldtse replied:
That is not allowed in the Metanorma ModSpec format. The existence of these links must have been a mistake because there is no point in referencing the table that this requirement is in. Please help change these into just the textual identifier, i.e. here `/conf/global`.

ronaldtse replied:
If it is citing another requirement, please state it as a `dependency`, not a `label`.

manuelfuenmayor replied:
Ok, thanks!

manuelfuenmayor replied:
Changes applied.

ronaldtse commented on standard/standard/clause_1_scope.adoc:
```diff
@@ -1,16 +1,40 @@
+appropriate to that technology.
 
-GeoPose Version 1.0 defines only one of many possible encoding methods for implementation of any or all of the eight STandardization Targets: JavaScript Object Notation (JSON). Each Standardization Target has a JSON-Schema:2019-9 definition. Most of the GeoPose standardization requirements are that concrete JSON GeoPose data objects shall conform to the corresponding JSON-Schema definition. The i ...
+GeoPose Version 1.0 defines only one of many possible encoding methods for
```
Let's not re-format the text when we don't need to.

manuelfuenmayor replied:
Not sure what you mean on this part, @ronaldtse.
What I see is that the indicated paragraph was just divided
into several lines as concurred.

ronaldtse replied:
I mean not to re-flow the text (do not break the paragraph into separate lines).

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,57 @@
+amendments to, or revisions of, any of these publications do not apply. For undated 
+references, the latest edition of the normative document referred to applies.
+
+* [[[json2017,JSON]]], Bray, T., Ed., "The JavaScript Object Notation (JSON) Data Interchange Format", STD 90, RFC 8259, December 2017, https://www.rfc-editor.org/info/std90
```
Use auto-fetch.

manuelfuenmayor replied:
Done.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,57 @@
+
+* [[[jsonschema2019,JSON-Schema 2019-09]]], H. Andrews, Ed. 2019. "JSON-Schema Specification", Draft 2019-09" https://json-schema.org/specification-links.html#draft-2019-09-formerly-known-as-draft-8
+
+* [[[jsonschemavalidation2019,JSON-Schema Validation 2019-09]]], A. Wright, Ed. 2019. "JSON-Schema Validation Recommendation", Draft 2019-09. https://tools.ietf.org/html/draft-handrews-json-schema-validation-02
```
Use auto-fetch.

manuelfuenmayor replied:
Done.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,57 @@
+
+* [[[collada,COLLADA]]], Khronos Group Inc.: COLLADA - Digital Asset Schema Release 1.5.0
+
+* [[[topic5,OGC Topic 5]]], OGC: The OpenGIS® Abstract Specification Topic 5: Features, OGC document 08-126
```
Use auto-fetch.

manuelfuenmayor replied:
Done.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,57 @@
+
+* [[[topic5,OGC Topic 5]]], OGC: The OpenGIS® Abstract Specification Topic 5: Features, OGC document 08-126
+
+* [[[topic8,OGC Topic 8]]], OGC: The OpenGIS™ Abstract Specification Topic 8: Relationships Between Features, OGC document 99-108r2
```
Use auto-fetch.

manuelfuenmayor replied:
Done

ronaldtse commented on standard/standard/clause_4_terms_and_definitions.adoc:
```diff
@@ -1,8 +1,16 @@
 == Terms and Definitions
-This document uses the terms defined in Sub-clause 5.3 of [OGC 06-121r8], which is based on the ISO/IEC Directives, Part 2, Rules for the structure and drafting of International Standards. In particular, the word “shall” (not “must”) is the verb form used to indicate a requirement to be strictly followed to conform to this standard.
```
Please use the default OGC boilerplate, do not use a custom one, we are trying to update to the latest OGC practices.

manuelfuenmayor replied:
Ok, fixed.
(Custom boilerplate was removed.)

ronaldtse commented on standard/standard/clause_5_conventions.adoc:
```diff
@@ -1,68 +1,86 @@
+[[fig3]]
 .Blue color Denotes Abstract or More General Classes
-image::./images/Blue_UML.png[align="center"]
+image::images/Blue_UML.png[align="center"]
```
Let's remove "align".

manuelfuenmayor replied:
`align` attribute removed.

ronaldtse commented on standard/standard/use_cases/use_cases.adoc:
```diff
@@ -1,18 +1,24 @@
 ===== Use Cases:
 
-[%header,cols="2,5,2"] 
+[%unnumbered,%header,cols="2,5,2"]
```
Why is this "unnumbered"?

manuelfuenmayor replied:
@ronaldtse that comes from original source as well.
Maybe they don't want these tables to get numbered?

ronaldtse replied:
I'm not seeing in the diff how the `unnumbered` part came from the original source. When the original source does not explicitly specify `unnumbered`, let's not add it. Tables are by default numbered in OGC except for very special circumstances.

manuelfuenmayor replied:
Huh, it turns out that the original markup doesn't include any `unnumbered` attribute, as you say.
I was looking at the PDF file, which shows the tables without any numbering.
Ok, I'm reverting that change.

manuelfuenmayor replied:
`unnumbered` attribute removed.

ronaldtse commented:
> Note: I kept the original form of the files and folders, although we usually have a separate subfolder for sections and one subfolder for images.

Let's do this as the last stage after PR approval so we can track the changes.

manuelfuenmayor committed:
Enable auto-fetch in bibliography identifiers
Remove anchor from Bibliography clause

manuelfuenmayor committed:
Enable auto-fetch

manuelfuenmayor committed:
Remove custom boilerplate from Terms and definitions

manuelfuenmayor committed:
Remove `align` attribute from images

ronaldtse commented on standard/standard/annex-bibliography.adoc:
```diff
@@ -1,38 +1,34 @@
+* [[[pose,8]]], Pose https://en.wikipedia.org/wiki/Pose_(computer_vision) 
 
-[9] OWL Time: https://www.w3.org/TR/2020/CR-owl-time-20200326/
+* [[[owl-time,9]]], OWL Time: https://www.w3.org/TR/2020/CR-owl-time-20200326/
```
Use auto-fetch with docid `W3C CR-owl-time-20200326`

manuelfuenmayor replied:
Done.

manuelfuenmayor committed:
Enable auto-fetch

manuelfuenmayor committed:
Remove cross-reference markup from label attribute in Requirements

manuelfuenmayor committed:
Remove `%unnumbered` attribute from tables

manuelfuenmayor committed:
Link Conformance Class with Requirements Classes

manuelfuenmayor committed:
Link Conformance test with Requirement

manuelfuenmayor commented:
@ronaldtse I have performed the corresponding links between[ Conformance Classes and Requirements Classes](https://github.com/metanorma/ogc-GeoPose/pull/2/commits/aa5bf14401b7f847eefd606cbec73b7577a68381).
However, there are several Conformance Classes that don't have their corresponding Requirements Class:
````
Requirements: (XML Line 004962): Conformance class test conf_class_basic_ypr_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005017): Conformance class test conf_basic_quaternion_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005070): Conformance class test conf_class_basic_quaternion_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005122): Conformance class test conf_class_advanced_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005174): Conformance class test conf_class_chain_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005225): Conformance class test conf_class_graph_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005277): Conformance class test conf_class_regular_series_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005329): Conformance class test conf_class_irregular_series_encoding_json has no corresponding Requirement class
Requirements: (XML Line 005381): Conformance class test conf_class_stream_encoding_json has no corresponding Requirement class

````

manuelfuenmayor commented:
@ronaldtse, I've found that Conformance tests are unable to link to Requirements that contain IDs.
More details in https://github.com/metanorma/metanorma-ogc/issues/394.

Can I remove the IDs from the requirements to solve the issue?
(E.g., to remove `id="req10"`, `id="req11"`, etc.)

ronaldtse commented:
> I've found that Conformance tests are unable to link to Requirements that contain IDs

What's an "ID"? What are these? `id="req10", id="req11"`?

ronaldtse commented:
> @ronaldtse I have performed the corresponding links between[ Conformance Classes and Requirements Classes](https://github.com/metanorma/ogc-GeoPose/pull/2/commits/aa5bf14401b7f847eefd606cbec73b7577a68381). However, there are several Conformance Classes that don't have their corresponding Requirements Class:
> 
> ```
> Requirements: (XML Line 004962): Conformance class test conf_class_basic_ypr_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005017): Conformance class test conf_basic_quaternion_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005070): Conformance class test conf_class_basic_quaternion_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005122): Conformance class test conf_class_advanced_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005174): Conformance class test conf_class_chain_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005225): Conformance class test conf_class_graph_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005277): Conformance class test conf_class_regular_series_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005329): Conformance class test conf_class_irregular_series_encoding_json has no corresponding Requirement class
> Requirements: (XML Line 005381): Conformance class test conf_class_stream_encoding_json has no corresponding Requirement class
> ```

Then this is a deficiency in the document. We will need to add the corresponding requirements classes as placeholders and let the authors fill them in.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,52 @@
-* [[ogc07-036, OGC 07-036]] OGC: OGC 07-036, OpenGIS® Geography Markup Language (GML) Encoding Standard, v3.2.1, 2007.
+* [[[json2017,RFC 8259]]]
+
+* [[[jsonschema2019,IETF draft-handrews-json-schema-validation-02]]], H. Andrews, Ed. 2019. "JSON-Schema Specification", Draft 2019-09" https://json-schema.org/specification-links.html#draft-2019-09-formerly-known-as-draft-8
```
This syntax is incorrect and causes this error:
```
[relaton-ietf] ("IETF draft-handrews-json-schema-validation-02") fetching...
[relaton-ietf] ("IETF draft-handrews-json-schema-validation-02") not found
```

See here for the correct syntax of citing Internet-Drafts. Please also **make sure** you compile before pushing.
https://github.com/relaton/relaton-ietf

```rb
# Fetch Internet-Draft document
RelatonIetf::IetfBibliography.get 'IETF I-D.draft-abarth-cake-01'
[relaton-ietf] ("IETF I-D.draft-abarth-cake-01") fetching...
[relaton-ietf] ("IETF I-D.draft-abarth-cake-01") found draft-abarth-cake-01
```

anermina replied:
Changed syntax.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,52 @@
+
+* [[[jsonschema2019,IETF draft-handrews-json-schema-validation-02]]], H. Andrews, Ed. 2019. "JSON-Schema Specification", Draft 2019-09" https://json-schema.org/specification-links.html#draft-2019-09-formerly-known-as-draft-8
+
+* [[[jsonschemavalidation2019,IETF draft-handrews-json-schema-validation-02]]], A. Wright, Ed. 2019. "JSON-Schema Validation Recommendation", Draft 2019-09. https://tools.ietf.org/html/draft-handrews-json-schema-validation-02
```
As above. Wrong syntax.

anermina replied:
Changed syntax.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,52 @@
+
+* [[[iso19157,ISO 19157:2013]]]
+
+* [[[iso19139,ISO/TC 211 19139:2007]]]
```
Wrong syntax. I understand that this may be the lack of understanding of the ISO document identifier syntax. It should have been "ISO 19139:2007".

anermina replied:
Changed syntax.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,52 @@
+
+* [[[topic8,OGC 99-108r2]]]
+
+* [[[ogc20-010, OGC 20-010]]], OGC: OGC 20-010, OGC City Geography Markup Language (CityGML) Part 1: Conceptual Model Standard, v3.0, 2021.
```
Prefer no space between comma and document identifier as a convention. Please post this to the editor's guide for future reference. Thanks.

anermina replied:
Removed extra spaces.

ronaldtse commented on standard/standard/clause_3_references.adoc:
```diff
@@ -1,32 +1,52 @@
+
+* [[[ogc20-010, OGC 20-010]]], OGC: OGC 20-010, OGC City Geography Markup Language (CityGML) Part 1: Conceptual Model Standard, v3.0, 2021.
+
+* [[[ogc07-036, OGC 07-036]]]
```
As above.

anermina replied:
Removed extra spaces.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+description:: Conformance with Global SDU requirements
+inherit:: None
+
+[conformance_test,id="rta1"]
```
Remove all `id="xxx"` because they are not necessary. We identify ModSpec instances using the `id` (also called `label`).

anermina replied:
Removed unnecessary `id` parameters.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+====
+[%metadata]
+label:: /conf/global
+subject:: <<rc_global,Requirement Class "Global">>
```
This should point to the URI that `rc_global` refers to, i.e.
```adoc
subject:: /req/global
```

anermina replied:
Applied.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+subject:: <<rc_global,Requirement Class "Global">>
+classification:: Target Type:SDU
+description:: Conformance with Global SDU requirements
+inherit:: None
```
`inherit::` is used to indicate what other conformance classes are a dependency for the current conformance class. In the case where there is none, do not specify.

Remove `inherit:: None`.

anermina replied:
Removed.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+======
+[%metadata]
+label:: /conf/target_dependency
+subject:: <<global,Requirement "Target Dependency">>
```
This should just be:
```adoc
subject:: /req/target_dependency
```

anermina replied:
Applied.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
-<s|Test Type|Conformance
-|===
-
+[conformance_test,id="rta1"]
```
Remove `id` here.

anermina replied:
Removed all `id`s.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+Standardization Targets.
+test-purpose:: Verify that this requirement is satisfied
+test-method:: Inspection
+reference:: <<global,/req/target_dependency>>
```
`reference::` is intended to refer to bibliographic items, not ModSpec URIs.

What we want here is actually `subject::` (in the future, `requirement::`).

Fix all of these to only give the URI, e.g.:
```adoc
subject:: /req/target_dependency
```

anermina replied:
Fixed.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+a rotation-only transformation using Yaw, Pitch, and Roll angles.
+inherit:: /conf/global
+
+[conformance_test,label="/conf/basic_ypr/sdu",name="B-YPR-SDU",id="rta4"]
```
These conformance tests are defined later. We should not re-define them here. This block should just be:
```adoc
conformance-test:: /conf/basic_ypr/sdu
```

Please change all the others too.

anermina replied:
Changes are applied.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+[%metadata]
+label:: /conf/target_dependency
+subject:: <<global,Requirement "Target Dependency">>
+name:: G-Target
```
The name is the "caption title". Implement this as a caption title.

anermina replied:
Added `name`s as block captions.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+test-purpose:: Verify that this requirement is satisfied
+test-method:: Inspection
+reference:: <<global,/req/target_dependency>>
+classification:: Test Type:Conformance
```
Throughout this document, `classification:: Test Type:Conformance` (and only this pattern) shows up.

`Test Type` is a superfluous element here because this block is a Conformance Test => it of course has "test type" of conformance.

Let's remove all of these lines throughout the document.

anermina replied:
Removed all `classification:: Test Type:Conformance`.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+====
+[%metadata]
+name:: G-Target
+label:: /conf/target_dependency
```
In the future we will call this `id::`. Keep for now.

ronaldtse commented on standard/standard/annex-ats.adoc:
```diff
@@ -1,1085 +1,1288 @@
+[conformance_test,id="rta1"]
+====
+[%metadata]
+name:: G-Target
```
Convert to caption title.

anermina replied:
Added `name`s as block captions.

ronaldtse commented on standard/standard/annex-glossary.adoc:
```diff
@@ -1,97 +1,132 @@
-The following terms and concepts have appeared in the discussion of the use cases and design of the GeoPose standard but not part of the terms defined in Terms and Definitions. These
+The following terms and concepts have appeared in the discussion of the use cases and
+design of the GeoPose standard but not part of the terms defined in Terms and
+Definitions. These
```
This paragraph is broken. This error originates from the original document.

ronaldtse commented on standard/standard/annex-glossary.adoc:
```diff
@@ -1,97 +1,132 @@
 
-*** *Acceleration:* 
-The time rate of change of velocity.
+* *Acceleration*: The time rate of change of velocity.
```
All terms in the OGC Annex Glossary must be formatted as Terms and definitions just like Clause 4 here.

anermina replied:
@ronaldtse Terms are currently not formatted as usually in Clause 4. Should I apply standard markup for terms and definitions in both Clause 4 and Glossary?

ronaldtse commented on standard/standard/clause_0_front_material.adoc:
```diff
@@ -1,89 +1,98 @@
-
-[#submission_contact_points,reftext='{table-caption} {counter:table-num}']
 .Submission Contact Points
-[width="100%",cols="5,10",options="header"]
```
Remove the `cols=...` as well. This is a semantic section.

anermina replied:
Removed.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+subject:: SDU
+description:: Global SDU Requirements
+inherit:: None
+requirement:: <<global,/req/target_dependency>>
```
Change to:
```
requirement:: /req/target_dependency
```

anermina replied:
Done.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+label:: /req/global
+subject:: SDU
+description:: Global SDU Requirements
+inherit:: None
```
Remove all "None" lines.

anermina replied:
Done.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-<s|Requirement G-LM|/req/lm_general 
-
-|===
+[requirement,type="class"]
```
Change to `[requirements_class]`.

anermina replied:
Changed.

anermina replied:
Changed.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+contents of these fields is outside the scope of GeoPose.
 
-===  Global Requirements (Normative)
+=== Global Requirements (Normative)
```
Remove the word `(Normative)`. All content not marked informative is normative.

anermina replied:
Removed.

anermina replied:
Removed all `(Normative)` indicators.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-
-|===
+[[req_class_basic_ypr_sdu]]
+[requirement,type="class"]
```
`[requirements_class]`

anermina replied:
Applied.

anermina replied:
Applied.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+inherit:: /req/global
+description:: Basic-Quaternion Logical Model SDU
+
+[requirement,name="B-TP-Lon-SDU",label="/req/tangent_plane_longitude/sdu",id="req1"]
+======
+======
```
This is just:
```adoc
requirement:: /req/tangent_plane_longitude/sdu
```

anermina replied:
Changed all requirements with unnecessary re-definitions.

anermina replied:
Applied.

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -94,1003 +133,1167 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+[requirement,id="req1"]
+====
+[%metadata]
+name:: B-TP-Lon-SDU
```
All `name::` should be encoded as a caption of the block.

anermina replied:
Added `name`s as block captions.

ronaldtse committed:
fix: metadata of standard

ronaldtse commented:
I've done a review. @metanorma/editors can you please help execute on the review recommendations?

Please also undo reflowing of text when it was not necessary. We want to retain the "most recent commit" history for the original authors for places where we did not actually change text.

ronaldtse commented:
> Unlike original, that includes the name of the requirement instead of the number: ![capture3](https://user-images.githubusercontent.com/33627611/178128659-acb1fc99-c066-4809-8e09-26a1a32f8171.PNG)

The original is incorrect. The name of the requirement exists but it is part of the block caption.

> In the case of "Requirement 4-2", I think there is an error from original source. The label should be `/req/time/geopose_instant/sdu` instead of `/req/geopose_instant/sdu`. To match with this requirement: ![capture4](https://user-images.githubusercontent.com/33627611/178128720-d9fa2d1f-7ef9-473f-a12f-1ec64b942fcf.PNG)
> 
> Maybe the field "time" was omitted by error. (I've corrected it.)

Thanks!

> In addition, I have applied a markup update to link some conformance tests to their respective requirements.

Thanks!

> This document is a draft, so there are some errors in the original source files. I've also applied some corrections:

> "Requirement" field in "Requirement S-FT-Element-SDU" and "Requirement S-Trailer-SDU" has been encoded as description - it seems to me there is an error in the original document, which shows description under ´Requirement´.
> I changed a typo in "Implemntation" to "Implementation" throughout the text.

Thanks!

ronaldtse commented:
@metanorma/editors I have also found additional typos throughout the document. Can you run a spell checker in the resulting document to fix them? Thanks.

anermina committed:
Remove faulty id parameters

anermina committed:
Correct typos

anermina committed:
Modify references for auto-fetching

anermina committed:
Remove extra spaces

anermina committed:
Fix subjects

anermina committed:
Fix URIs in subjects

anermina committed:
Remove redefining items and apply 
eference_class identifier

anermina committed:
Remove cross-references from URIs

anermina committed:
Remove classification in conformance tests

anermina committed:
Remove normative indicators

anermina committed:
Remove cols attribute from the front material

anermina committed:
Remove added linebreaks

anermina committed:
Add
ame attributes as block captions

anermina committed:
Adjust links in requirements/conformance classes

anermina committed:
Modify cross-references and anchors to avoid broken links

anermina commented:
Most of the requested changes have been applied. Spelling mistakes/typos have been corrected and reflowing of the text has been undone.

However, I would like to add a few notes:
1. While converting the document, I've noticed there are some content differences between the PDF which was already generated (original PDF) and source .adoc file which was also already generated. I assume this has been made intentionally, since .adoc is used for creating a new document version.
Example: In Section 2, PDF has an additional paragraph which wasn't (and isn't) present in .adoc.
![image](https://user-images.githubusercontent.com/55683568/181093486-d89ac955-4e4e-4d7f-865d-3bb06469af6e.png)
2. Some requirement classes and requirements have the same label, which seems wrong to me. Example: `/req/basic_ypr/sdu`.
I removed non-existing requirements with the same name as the requirements class from that requirements class.
3. In the https://github.com/metanorma/ogc-GeoPose/pull/2/commits/0db85e56ab5b773121fc161fc46df0f9371b4c51, I modified requirements in requirements classes to match the labels of the existing requirements.
4. Terms Frame and Euclidean Space are not defined in the document, but there were cross-references pointing to them. I removed the cross-references.
5. I've noticed both "color" and "colour" in the text. Is this document written in American English or British English?

manuelfuenmayor commented:
In addition to @anermina comments:

> This should point to the URI that `rc_global` refers to, i.e.
> 
> ```
> subject:: /req/global
> ```

@ronaldtse, currently this approach does not link the Conformance Classes to their corresponding Requirement Classes.
I.e., we are still getting a bunch of compilation errors in this regard. More details in https://github.com/metanorma/metanorma-ogc/issues/401

> @ronaldtse Terms are currently not formatted as usually in Clause 4...

Yes, and moreover, it has a custom boilerplate. To show a part of it:

```adoc
The GeoPose Conceptual Model(CM) consists of linked definitions of
terms denoting concepts expressed in the GeoPose LM and structural
data unit(SDU) specifications for the standardization targets. The CM
describes a (non-normative) domain of discourse for terms used in
defining a precise and normative Logical Model(LM) expressed as a
Unified Modelling Language (UML) class diagram.

...
````

Should we use this boilerplate or the default provided by Metanorma?

ronaldtse committed:
chore: remove unused conformance class/test anchors

ronaldtse committed:
feat: update glossary format and refine definitions

anermina committed:
Modify Terms and Definitions and apply smart cross-referencing

manuelfuenmayor committed:
Re-structure SDU Global Requirements

manuelfuenmayor committed:
Re-structure SDU Requirements

manuelfuenmayor committed:
Re-structure SDU Requirements

manuelfuenmayor committed:
Re-structure SDU Requirements

manuelfuenmayor committed:
Re-structure SDU Requirements

manuelfuenmayor committed:
Re-structure Basic Requirements

manuelfuenmayor committed:
Re-structure Target Requirements

manuelfuenmayor committed:
Re-structure Requirements in "Composite" folder

manuelfuenmayor committed:
Re-structure Requirements

manuelfuenmayor committed:
Place include attributes for Requirements

manuelfuenmayor committed:
Relocate Requirement adoc files

manuelfuenmayor committed:
Re-structure Global Requirement Class

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -77,8 +73,7 @@ Classes, attributes, and relationships of the GeoPose domain are specified in a
 SDUs provide Standardization Targets that are independent of serialization/encoding format. This allows multiple equivalent serializations to be defined. Each SDU that may be expressed as a concrete data object is associated with a corresponding element (class or attribute) in the logical model.
 
 [[standardization_targets]]
```
This anchor does not point to any reference. Do we really need it?

ronaldtse replied:
Let’s get rid of it.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +208,429 @@ The Graph Target supports a network of object relative poses. The graph is a dir
 
 [[req_graph]]
```
This anchor does not have a reference.

ronaldtse replied:
Let’s get rid of it.

manuelfuenmayor commented on standard/standard/annex-glossary.adoc:
```diff
@@ -1,97 +1,231 @@
-[appendix]
-:appendix-caption: Annex 
+[[annex-glossary]]
+[appendix,obligation=informative]
 == Glossary
```
@ronaldtse, according to the [documentation,](https://www.metanorma.org/author/ogc/authoring-guide/terms-definitions/#terminology-clauses) this attribute: `[heading='terms and definitions']`, should be added.
Though it seems that it doesn't make any difference to the current result.

ronaldtse replied:
Maybe the keyword “Glossary” is now supported officially? Ping @opoudjis   if so we should fix the documentation. Thanks!

ronaldtse replied:
@manuelfuenmayor no, this is incorrect. According to documentation, we do not need an additional `heading`:
<img width="776" alt="Screen Shot 2022-08-15 at 1 05 10 PM" src="https://user-images.githubusercontent.com/11865/184579485-5d7cda6a-4eef-42f1-b4f1-46fc054b91a0.png">

Just `[appendix]` and `== Glossary` is sufficient.

manuelfuenmayor committed:
Correct filename of Requirements Class

ronaldtse commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -125,28 +125,7 @@ image::diagram/Basic-YPR_SDU.PNG[]
 
 *Requirements Class*
```
Remove this line, the labels "Requirements class", "Requirement" etc. are automatically given.

manuelfuenmayor replied:
Done.

ronaldtse commented on standard/standard/requirements/requirements_class/RC_Global.adoc:
```diff
@@ -0,0 +1,22 @@
+[[req_class_basic_ypr_sdu]]
```
Given that the ID of this requirements class is `/req/basic_ypr/sdu`, let's name this file path `requirements/basic_ypr/sdu.adoc`.

Individual requirements can be: `requirements/basic_ypr/sdu/xxx.adoc`.

ronaldtse replied:
@manuelfuenmayor please help implement this new path pattern for all `[requirement]`, `[requirements_class]`, `[conformance_class]` and `[abstract_test]` blocks.

manuelfuenmayor committed:
Remove "*Requirements Class*" text

ronaldtse committed:
chore: remove unnecessary temp files

ronaldtse committed:
fix: typo "addtionalProperties" should be "additionalProperties"

manuelfuenmayor committed:
Revert "Correct filename of Requirements Class"

This reverts commit 89707a95b5249fd98b6bf1543b92b01b7ebd613d.

manuelfuenmayor committed:
Revert "Re-structure Requirements"

This reverts commit 165fa90389ce56504365766c31b4841eaf86ba41.

manuelfuenmayor commented:
@ronaldtse, this document is getting even more confusing.
In "clause_7_normative_text.adoc" file, there are Requirements with the same ID (label) but with different descriptions.

For example, there is:
```adoc
[[req_tangent_plane_h]]
[requirement]
====
[%metadata]
label:: /req/tangent_plane_h/sdu
description:: An instance of a GeoPose tangentPoint.h attribute SHALL be
expressed as a height in meters above the WGS-84 ellipsoid and represented
as a signed real number value.
====
````

and later in another sub-section, there is:

```adoc
[requirement]
.B-TP-h-SDU
====
[%metadata]
label:: /req/tangent_plane_h/sdu
description:: An instance of a GeoPose tangentPoint.h attribute SHALL be
expressed as a height in meters above the WGS-84 ellipsoid, represented
as a signed real number. If the tangent point is above the WGS-84 ellipsoid,
the value SHALL be positive. If the tangent point is below the WGS-84 ellipsoid,
the value SHALL be negative.
====
````

Which one should we select?

manuelfuenmayor committed:
Remove `*Requirement Class*` line

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+
+
+[[req_class_basic_ypr_sdu]]
+[requirements_class]
+====
+[%metadata]
+label:: /req/basic_ypr/sdu
+subject:: Implementation of Logical Model
+description:: Basic-YPR Logical Model SDU
+inherit:: /req/global
+requirement:: /req/basic_ypr/sdu
```
The label of this Requirements Class (`/req/basic_ypr/sdu`) is duplicated in one of its Requirements.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-2+<|An implementation of a Basic-YPR Target SHALL consist of an Outer Frame specified by an implicit WGS-84 CRS and an implicit EPSG 4461-CS (LTP-ENU) coordinate system and explicit parameters to define the tangent point. The Inner Frame SHALL be a rotation-only transformation using Yaw, Pitch, and Roll angles specified in degrees and represented as signed real number values.
-
-|===
+[requirement]
+====
+[%metadata]
+label:: /req/basic_ypr/sdu
```
As described above, this Requirement is the one that has the same label as its Requirements Class.

ronaldtse replied:
Exactly, this is wrong. We need to fix these through the new path pattern.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-
-|===
-
+[requirement]
+====
+[%metadata]
+label:: /req/tangent_plane_longitude/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-2+<|An instance of GeoPose tangentPoint.latitude attribute SHALL be expressed as decimal degrees and represented as signed real number values.
-
-|===
+[requirement]
+====
+[%metadata]
+label:: /req/tangent_plane_latitude/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-2+<|An instance of a GeoPose tangentPoint.h attribute SHALL be expressed as a height in meters above the WGS-84 ellipsoid and represented as a signed real number value.
-
-|===
+[requirement]
+====
+[%metadata]
+label:: /req/tangent_plane_h/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+requirement:: /req/orientation_quaternion/sdu
+====
+
+[requirement]
+.B-TP-Lon-SDU
+====
+[%metadata]
+label:: /req/tangent_plane_longitude/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+description:: An instance of a GeoPose tangentPoint.longitude attribute SHALL be expressed as decimal degrees, represented as a signed real number.
+====
+
+[requirement]
+.B-TP-Lat-SDU
+====
+[%metadata]
+label:: /req/tangent_plane_latitude/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
+description:: An instance of GeoPose tangentPoint.latitude attribute SHALL be expressed as decimal degrees, represented as a signed real number.
+====
+
+[requirement]
+.B-TP-h-SDU
+====
+[%metadata]
+label:: /req/tangent_plane_h/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-
-<<<
-
+[requirement]
+.B-Quaternion-SDU
+====
+[%metadata]
+label:: /req/orientation_quaternion/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-<s|ID |<<req_time_geopose_instant_sdu,/req/time/geopose_instant/sdu>>
-2+<|The Logical Model attribute GeoPoseInstant SHALL be expressed in Unix Time in seconds multiplied by 1,000. The unit of measure SHALL be milliseconds.
-|===
+[requirement]
+.GP-Instant-SDU
+====
+[%metadata]
+label:: /req/time/geopose_instant/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-<s|ID |<<req_sdu_frame_specification_authority,/req/frame_specification/authority/sdu>>
-2+<|The frame_specification.authority attribute SHALL contain a string uniquely specifying a source of reference frame specifications.
-|===
+[requirement]
+.FS-Authority-SDU
+====
+[%metadata]
+label:: /req/frame_specification/authority/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-<s|ID |<<req_sdu_frame_specification_id,/req/frame_specification/id/sdu>>
-2+<|The frame_specification.ID attribute SHALL be a string uniquely defining a frame within the authority.
-|===
+[requirement]
+.FS-ID-SDU
+====
+[%metadata]
+label:: /req/frame_specification/id/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -89,305 +84,261 @@ The Basic and Advanced Standardization Targets differ in the level of options an
-<<<
-
-<<<
+[requirement]
+.FS-Parameters-SDU
+====
+[%metadata]
+label:: /req/frame_specification/parameters/sdu
+description:: The frame_specification.parameter attribute SHALL contain all parameters needed for the corresponding authority and ID. The definition of these parameters is outside the scope of GeoPose.
+====
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+requirement:: /req/transform_index/sdu
+====
+
+[requirement]
+.Graph-SDU
+====
+[%metadata]
+label:: /req/time/geopose_instant/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The Logical Model attribute GeoPoseInstant SHALL be expressed in Unix Time in seconds multiplied by 1,000. The unit of measure SHALL be milliseconds.
+====
+
+[requirement]
+.FS-Authority-SDU
+====
+[%metadata]
+label:: /req/frame_specification/authority/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.authority attribute SHALL contain a string uniquely specifying a source of reference frame specifications.
+====
+
+[requirement]
+.FS-ID-SDU
+====
+[%metadata]
+label:: /req/frame_specification/id/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.ID attribute SHALL be a string uniquely defining a frame within the authority.
+====
+
+[requirement]
+.FS-Parameters-SDU
+====
+[%metadata]
+label:: /req/frame_specification/parameters/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+requirement:: /req/frame_specification/parameters/sdu
+====
+
+[requirement]
+.GP-Instant-SDU
+====
+[%metadata]
+label:: /req/time/geopose_instant/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The Logical Model attribute GeoPoseInstant SHALL be expressed in Unix Time in seconds multiplied by 1,000. The unit of measure SHALL be milliseconds.
+====
+
+[requirement]
+.FS-Authority-SDU
+====
+[%metadata]
+label:: /req/frame_specification/authority/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.authority attribute SHALL contain a string uniquely specifying a source of reference frame specifications.
+====
+
+[requirement]
+.FS-ID-SDU
+====
+[%metadata]
+label:: /req/frame_specification/id/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.ID attribute SHALL be a string uniquely defining a frame within the authority.
+====
+
+[requirement]
+.FS-Parameters-SDU
+====
+[%metadata]
+label:: /req/frame_specification/parameters/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: An interposeDuration property SHALL be implemented as a GeoPoseDuration.
+====
+
+[requirement]
+.FS-Authority-SDU
+====
+[%metadata]
+label:: /req/frame_specification/authority/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.authority attribute SHALL contain a string uniquely specifying a source of reference frame specifications.
+====
+
+[requirement]
+.FS-ID-SDU
+====
+[%metadata]
+label:: /req/frame_specification/id/sdu
```
This Requirement is duplicated.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.ID attribute SHALL be a string uniquely defining a frame within the authority.
+====
+
+[requirement]
+.FS-Parameters-SDU
+====
+[%metadata]
+label:: /req/frame_specification/parameters/sdu
```
This Requirement is duplicated.

ronaldtse replied:
All duplicated requirements must be removed. Only keep the first instance.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: A TransitionModel property SHALL be implemented as one of the values in the TransitionModel enumeration.
+====
+
+[requirement]
+.FS-Authority-SDU
+====
+[%metadata]
+label:: /req/frame_specification/authority/sdu
```
This Requirement is duplicated.

ronaldtse replied:
All duplicated requirements must be removed. Only keep the first instance.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.authority attribute SHALL contain a string uniquely specifying a source of reference frame specifications.
+====
+
+[requirement]
+.FS-ID-SDU
+====
+[%metadata]
+label:: /req/frame_specification/id/sdu
```
This Requirement is duplicated.

ronaldtse replied:
Removed.

manuelfuenmayor commented on standard/standard/clause_7_normative_text.adoc:
```diff
@@ -397,700 +348,669 @@ The Graph Target supports a network of object relative poses. The graph is a dir
+description:: The frame_specification.ID attribute SHALL be a string uniquely defining a frame within the authority.
+====
+
+[requirement]
+.FS-Parameters-SDU
+====
+[%metadata]
+label:: /req/frame_specification/parameters/sdu
```
This Requirement is duplicated.

ronaldtse replied:
All duplicated requirements must be removed. Only keep the first instance.

manuelfuenmayor reviewed (commented):
@ronaldtse, I have marked the position of all the duplicated Requirements.

manuelfuenmayor committed:
Remove unnecessary anchors

ghobona commented:
The abstract tests are being displayed with the label "Requirement test". 

![Screenshot 2022-08-08 at 10 51 12](https://user-images.githubusercontent.com/18121038/183391217-ae866153-e16d-4939-9f06-1267851bc818.png)


The label should be "Abstract test" just like in https://docs.ogc.org/is/18-062r2/18-062r2.html#toc61

ghobona commented:
Please wait until I provide you with a crosswalk before fixing the "Requirement test -> Abstract test" labelling. I will provide you with the crosswalk by 2022-08-14.

ghobona commented:
I have agreed with the Editor on the mappings/crosswalk. I will complete the editing of the document.

Please proceed to merge the changes from [metanorma/ogc-GeoPose/tree/geopose_encoding](https://github.com/metanorma/ogc-GeoPose/tree/geopose_encoding) to [metanorma/ogc-GeoPose/tree/main](https://github.com/metanorma/ogc-GeoPose/tree/main). Once you have merged the [PR#2](https://github.com/metanorma/ogc-GeoPose/pull/2),  please proceed to create another PR for updating [opengeospatial/GeoPose/tree/main](https://github.com/opengeospatial/GeoPose/tree/main).

ronaldtse committed:
fix: fix whitespacing issues between words

ronaldtse committed:
fix: correct Submitters section syntax

ronaldtse committed:
fix: clean up Glossary encoding, and remove `noref` (because we want references!)

manuelfuenmayor committed:
Remove "noref" option from concepts

manuelfuenmayor committed:
Remove clarification about formatting convention in Terms and definitions clause

ronaldtse committed:
feat: clean up terms and definitions, requirements and content

ronaldtse committed:
feat: fix duplicated requirements and broken modspec links

ronaldtse committed:
feat: fix duplicated requirements and broken modspec links

manuelfuenmayor committed:
Add "req" folder containing the Requirements

manuelfuenmayor committed:
Replace Requirement encodings with include directives

manuelfuenmayor committed:
Add "conf" folder containing the Conformance Tests

manuelfuenmayor committed:
Replace Conformance Test encodings with include directives

manuelfuenmayor committed:
Add EDITOR comment

manuelfuenmayor commented:
@ronaldtse, I have splitted the Requirements and Conformance Tests into adoc files.
Requirements are located into *req* folder, and Conformance Tests into *conf* folder.
There is caveat, there are cases where a Conformance Class has the same ID of an Abstract Test. In that case, I appended the `_class.adoc` to the filename of the Conformance Class in order to solve the conflict. (Same approach is taken for Requirements.)

manuelfuenmayor commented:
@ronaldtse, I have splitted the Requirements and Conformance Tests into adoc files.
Requirements are located into *req* folder, and Conformance Tests into *conf* folder.
There is caveat, there are cases where a Conformance Class has the same ID of an Abstract Test. In that case, I appended the `_class.adoc` to the filename of the Conformance Class in order to solve the conflict. (Same approach is taken to Requirements.)

opoudjis commented:
> The abstract tests are being displayed with the label "Requirement test".
> 
> ![Screenshot 2022-08-08 at 10 51 12](https://user-images.githubusercontent.com/18121038/183391217-ae866153-e16d-4939-9f06-1267851bc818.png)
> 
> The label should be "Abstract test" just like in https://docs.ogc.org/is/18-062r2/18-062r2.html#toc61

I cannot replicate this issue. (Note that some debugging has taken place over the last fortnight)

<img width="852" alt="Στιγμιότυπο 2022-08-16, 20 45 58" src="https://user-images.githubusercontent.com/1911039/184861697-11e9f33b-654b-4f84-b9b6-8b3c955ab697.png">

ronaldtse committed:
feat: fully migrate requirements to modspec and remove old instances

ronaldtse committed:
feat: merge use_cases into clause 6

ronaldtse committed:
feat: fix terms referencing

ronaldtse committed:
feat: remove placeholder and obsolete files

ronaldtse committed:
fix: fix readme formatting

ronaldtse committed:
fix: use bibliographic references

ronaldtse committed:
feat: add Gemfile

ronaldtse committed:
feat: only keeping one set of generated artifacts under standard/, removing standard/standard/**/geopose_standard.*

ronaldtse committed:
feat: remove unused content seemingly already integrated into document

ronaldtse committed:
feat: add requirements-model specification to ogc

ronaldtse commented:
I have done a full consolidation on the ModSpec instances. Now with the new path-based files used to specify ModSpec instances it is very clear what is included and what is missing.

We have not yet performed a full validation on the linkage of ModSpec instances (pending https://github.com/metanorma/mn-requirements/issues/3).

ronaldtse committed:
feat: add GHA

ronaldtse committed:
fix: req/time/instant

ronaldtse committed:
fix: req/time/duration

ronaldtse committed:
fix: req/series/irregular

ronaldtse committed:
fix: req/irregular_series/encoding/json

ronaldtse committed:
fix: req/basic_ypr/target, req/basic_quaternion/target

ronaldtse committed:
Update generate.yml

ronaldtse committed:
fix: update conformance tests to use "requirement:" to "target:"

manuelfuenmayor committed:
Fix unresolved term and include directives

manuelfuenmayor committed:
Remove dimensions in figure directive

manuelfuenmayor commented:
@ronaldtse I have checked the document and reported a couple of issues.
Hopefully, these are the remaining ones. Maybe @anermina can help us
confirm when she has time 👍.

opoudjis committed:
markup update on requirement identifiers: https://github.com/metanorma/metanorma-ogc/issues/414

ronaldtse commented:
There are some cross-reference errors I cannot work out why they exist. @manuelfuenmayor @anermina any ideas?

```
No label has been processed for ID _5543b8dd-2af0-8dea-1165-aa1b4cfeb489
No label has been processed for ID _ff223c88-5d90-07b4-ea0b-f5223c6e53ac
No label has been processed for ID _4f00095e-e78d-6d3f-1918-fb1b60f8fb17
No label has been processed for ID _813b2e6c-216a-4054-a8b8-7df92c648404
No label has been processed for ID _be179805-40a8-81b7-b4be-a2097348d0b1
No label has been processed for ID _b511aaa5-909d-395d-f2fe-30c114dc8c0f
No label has been processed for ID _3e5f1c56-9cdf-5b37-7601-e6a5cf0cfbe1
No label has been processed for ID _9397a0b5-bfb2-b9ec-f546-22c7109e0568
No label has been processed for ID _3946c592-1491-43f3-6569-2d1dce665f56
No label has been processed for ID _274dbde5-816c-aa28-4df6-ceafd7dd1700
No label has been processed for ID _dc216b7f-f01e-2af9-d9ec-7cb8191b22e9
No label has been processed for ID _fe26b0e2-039a-1dd9-85f1-5dafc015ffd6
No label has been processed for ID _9aa6ee87-2e8b-6713-2693-44630d0a3fc0
No label has been processed for ID _c55f3b3a-c5ad-3951-e8b1-351059b7db3f
No label has been processed for ID _9ee736bf-dc8e-f1ca-6c18-d71dea8113f1
No label has been processed for ID _08e43d5e-ad95-7ef6-a793-b5c7bd0b6988
No label has been processed for ID _bcc493eb-2f5f-e08c-5820-3e13f6782972
No label has been processed for ID _e7f0269b-20a1-75bd-07a3-ea6a5656ea17
No label has been processed for ID _3fd486c4-97e1-e19a-27e3-70b24c3861cb
No label has been processed for ID _635f0125-a10b-1786-e9a9-c11ac1250830
No label has been processed for ID _e029eb0a-28ea-f6db-45be-828778fc9886
No label has been processed for ID _f1789bd5-d1db-6cd3-2d05-ecd9d303dd26
No label has been processed for ID _aa999170-6bf1-97d6-35f9-acfe698b95e6
No label has been processed for ID _dbfd0bb0-7ba2-47bb-b719-ac2c12895034
No label has been processed for ID _fc138237-2c31-6ab6-70b4-a3034d9ec267
No label has been processed for ID _33b53c2f-8344-6af2-8ce4-51dadbfe77ed
No label has been processed for ID _5b313d47-0ed8-13b5-da09-811031872891
No label has been processed for ID _d4db4831-0245-25d1-6460-880316f7885c
No label has been processed for ID _7d771733-89bd-012c-9c69-34dfa371aa13
No label has been processed for ID _ad096d77-a589-b59c-3335-caf1d7529768
No label has been processed for ID _bc291e9b-45cd-33b9-2b5a-119cf1e5a2b0
No label has been processed for ID _8708854c-2cea-1a61-4143-6cd5fd0468c6
No label has been processed for ID _15915a13-e9c9-e507-70f4-d2a32213bc90
No label has been processed for ID _4855beec-ff52-3b9b-e8c2-9b301ab4bc7f
No label has been processed for ID _0fe5c17c-9164-e994-33a7-0035c75f7c41
No label has been processed for ID _fd9aaa73-8e71-b3cf-e711-76ad4b18ad80
No label has been processed for ID _0725ed1f-fe6c-8574-1604-913af1501152
No label has been processed for ID _19970ba3-8dae-d7f6-4754-219b0c6a017a
No label has been processed for ID _76571d85-d6ff-1a50-f56a-773764fd2825
No label has been processed for ID _0e6e7400-ec16-bf06-bd80-40628337626c
No label has been processed for ID _6f312035-87b3-597c-67b2-52e2bf719c51
No label has been processed for ID _83425a81-da87-ab38-69d5-665c594969dd
No label has been processed for ID _91a4d9e9-a618-4b46-3ffa-804171db9807
No label has been processed for ID _40128b4a-2c37-6f8d-5213-cb0ab80d75a6
No label has been processed for ID _c0e892d3-2104-37d9-a311-0cb115ba5af6
No label has been processed for ID figure-1
No label has been processed for ID core_logical_model
No label has been processed for ID time_logical_model
No label has been processed for ID sequence_logical_model
No label has been processed for ID _1057d44c-06d9-6bb9-33d4-99755af37cc5
No label has been processed for ID basic_ypr_json_schema
No label has been processed for ID basic_quaternion_strict_json_schema
No label has been processed for ID basic_quaternion_permissive_json_schema
No label has been processed for ID advanced_geopose_json_schema
No label has been processed for ID graph_json_schema
No label has been processed for ID chain_json_schema
No label has been processed for ID regular_series_json_schema
No label has been processed for ID streamheader_json_schema
No label has been processed for ID streamelement_json_schema
No label has been processed for ID streamrecord_json_schema
No label has been processed for ID _3eb1edb6-0d54-6519-1abe-8ec219ef015c
No label has been processed for ID _98ad5a1a-3192-a622-72c2-c3e5d934aa82
No label has been processed for ID _8b8e677e-6048-116c-dc85-cc70a3b572c4
No label has been processed for ID _004be9ae-0f1c-fc41-cf32-3ea4e67e120b
```

ronaldtse commented:
For example, `core_logical_model` is definitely defined:
```adoc
==== Core

The Logical Model Core contains the essential elements specific to the GeoPose modelled as a transformation between an anchoring Outer Frame and one or more derived Inner Frames. This is described in <<core_logical_model>>.

[[core_logical_model]]
.Core logical model
image::models/Logical_Core.png[]
```

ronaldtse commented:
> For example, `core_logical_model` is definitely defined:

This issue is fixed but not yet released.

ronaldtse committed:
feat: fix a bunch of unlinked/misnamed req/confs

ronaldtse committed:
fix: fix conf/orientation/quaternion

opoudjis committed:
markup fixes

opoudjis committed:
Merge branch 'geopose_encoding' of https://github.com/metanorma/ogc-GeoPose into geopose_encoding

ronaldtse committed:
feat: fix more modspec linking

ronaldtse commented:
Let's get this merged first.
